storagenode/storagenodedb/pieceinfo.go

```diff
@@ -47,6 +47,31 @@ func (db *pieceinfo) Add(ctx context.Context, info *pieces.Info) (err error) {
 	return ErrInfo.Wrap(err)
 }
 
+// GetPieceIDs gets pieceIDs using the satelliteID
+func (db *pieceinfo) GetPieceIDs(ctx context.Context, satelliteID storj.NodeID, createdBefore time.Time, limit, offset int) (pieceIDs []storj.PieceID, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	rows, err := db.db.QueryContext(ctx, db.Rebind(`
+		SELECT piece_id
+		FROM pieceinfo
+		WHERE satellite_id = ? AND piece_creation < ?
+		LIMIT ? OFFSET ?
```

Comment:
ORDER BY is usually required to make sure that LIMIT and OFFSET are applied coherently.

Reply:
Also worth mentioning is that LIMIT and OFFSET doesn't always work all that well outside of a transaction.  Between two calls to GetPieceIDs(), a DELETE could cause you get the same Piece again.  An INSERT could cause you to skip Pieces.